File: UCMS.Models/DbModels/SiteContentCreation/DocumentType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UCMS.Models.Interface;

namespace UCMS.Models.DbModels.SiteContentCreation
{
    public class DocumentType : IEntity
    {
        public int Id { get => DocumentTypeId; }

        public int DocumentTypeId { get; set; }
        public string Name { get; set; } = string.Empty;

        public int SiteId { get; set; }
        public Site Site { get; set; } = null!;

        public Code? Code { get; set; }

        public ICollection<Property> Properties { get; set; } = [];
        public ICollection<Content> Contents { get; set; } = [];
    }
}

File: UCMS.FrontendReactVite/src/models/siteContentCreation/DocumentType.ts
import type { Code } from "./Code";
import type { Content } from "./Content";
import type { Property } from "./Property";
import type { Site } from "./Site";

export interface DocumentType {
    documentTypeId: number,
    name: string,

    siteId: number,
    site: Site | null,

    properties: Property[],
    contents: Content[],
    code: Code,
};